protocol/xmrmaker/instance.go

```diff
@@ -57,15 +60,126 @@ func NewInstance(cfg *Config) (*Instance, error) {
 		go cfg.Network.Advertise()
 	}
 
-	return &Instance{
+	inst := &Instance{
 		backend:        cfg.Backend,
 		dataDir:        cfg.DataDir,
 		walletFile:     cfg.WalletFile,
 		walletPassword: cfg.WalletPassword,
 		offerManager:   om,
 		swapStates:     make(map[types.Hash]*swapState),
 		net:            cfg.Network,
-	}, nil
+	}
+
+	err = inst.checkForOngoingSwaps()
+	if err != nil {
+		return nil, err
+	}
+
+	return inst, nil
+}
+
+func (b *Instance) checkForOngoingSwaps() error {
+	swaps, err := b.backend.SwapManager().GetOngoingSwaps()
+	if err != nil {
+		return err
+	}
+
+	for _, s := range swaps {
+		if s.Provides != types.ProvidesXMR {
+			continue
+		}
+
+		if s.Status == types.KeysExchanged || s.Status == types.ExpectingKeys {
+			// TODO: set status to aborted, delete info from recovery db
```

Comment:
Is that a lot of code? Any reason to not put it in this PR?

Reply:
I forgot lol, will add!